src/amg/azext_amg/custom.py

```diff
@@ -25,55 +28,108 @@
-
-    logger.warning("Grafana instance of '%s' was created. Now creating default role assignments for its "
-                   "managed identity, and current CLI account unless --principal-ids are provided", grafana_name)
-
-    subscription_scope = '/subscriptions/' + client._config.subscription_id  # pylint: disable=protected-access
-
-    if not principal_ids:
-        user_principal_id, user_principal_type = _get_login_account_principal_id(cmd.cli_ctx)
-        principal_ids = [user_principal_id]
-        principal_types = [user_principal_type]
-    grafana_admin_role_id = resolve_role_id(cmd.cli_ctx, "Grafana Admin", subscription_scope)
-
-    for p, t in zip(principal_ids, principal_types):
-        _create_role_assignment(cmd.cli_ctx, p, t, grafana_admin_role_id, resource.id)
-
-    if resource.identity:
-        monitoring_reader_role_id = resolve_role_id(cmd.cli_ctx, "Monitoring Reader", subscription_scope)
-        _create_role_assignment(cmd.cli_ctx, resource.identity.principal_id, "ServicePrincipal",
-                                monitoring_reader_role_id, subscription_scope)
-
-    return resource
+class GrafanaCreate(_GrafanaCreate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema['skip_system_assigned_identity'] = AAZStrArg(
+            options=['--skip-system-assigned-identity', '--skip-identity'],
+            help='Do not enable system assigned identity. Use this option if you want to manage the identity '
+                 'yourself.',
+        )
+        args_schema['skip_role_assignments'] = AAZStrArg(
+            options=['--skip-role-assignments'],
+            help='Skip creating default role assignments for the managed identity of the Grafana instance and '
+                 'the current CLI account. Use this option if you want to manage role assignments yourself.',
+        )
+        args_schema['principal_ids'] = AAZListArg(
+            options=['--principal-ids'],
+            help='Space-separated Azure AD object ids for users, groups, etc. to be made as Grafana Admins. '
+                 'Once provided, CLI won\'t make the current logged-in user as Grafana Admin',
+        )
+        args_schema['principal_ids'].Element = AAZStrArg()
+        args_schema['identity']._registered = False  # pylint: disable=protected-access
+
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if args.skip_role_assignments and args.principal_ids:
+            raise ArgumentUsageError("--skip-role-assignments | --assignee-object-ids")
+
+        if not args.skip_system_assigned_identity:
+            args.identity = {"type": "SystemAssigned"}
+
+        process_grafana_create_namespace(self.ctx, self.ctx.args)
+
+    # override the output method to create role assignments after instance creation
+    def _output(self, *args, **kwargs):
```

Comment:
It should be in `post_operations` function instead of `_output`

Reply:
Unfortunately `post_operations` does not have ``self.ctx.vars.instance`` available to reference, and `post_instance_create`/`post_instance_update`, which does have that, is only available to override in subcommands. I found this way to achieve what was needed while still using the recommended inheritance method of customization. Is this acceptable?

Reply:
OK, that's fine. It's because it's a LRO and the instance will created only when LRO pulling finished.